tests/test_public_api.py

```diff
@@ -80,11 +80,17 @@ def test_parse_path_user_proj(mock_server, api):
     assert r == "run"
 
 
-def test_parse_path_proj(mock_server, api):
-    u, p, r = api._parse_path("proj")
+def test_parse_path_run(mock_server, api):
+    u, p, r = api._parse_path("run")
     assert u == "mock_server_entity"
-    assert p == "proj"
-    assert r == "proj"
+    assert p == "test"
+    assert r == "run"
```

Comment:
> I think we want a test that ensures we pull the entity / project from the environment if it's set...

@vanpelt That's what this test ensures. `"test"` is set in `api.settings`.

What I need clarification on is the expected behaviour for when project _isn't_ set and we try to parse a path with only the `run_id`. The existing test assumes that a single entry in the path is the project name but that doesn't match up with the [documentation](https://docs.wandb.ai/ref/python/public-api/api#run):

> path to run in the form `entity/project/run_id`. If `api.entity` is set, this can be in the form `project/run_id` and if `api.project` is set this can just be the `run_id`